src/pcre2test.c

```diff
@@ -5337,7 +5350,7 @@ if ((pat_patctl.control & CTL_INFO) != 0)
     for (; namecount > 0; namecount--)
       {
       int imm2_size = test_mode == PCRE8_MODE ? 2 : 1;
-      size_t length = STRLEN(nametable + imm2_size);
```

Comment:
This was really surprising to me.

`nametable + imm2_size` is really not valid, because `nametable` has type `void*`. And yet, due to the macro, it was fortuitously expanding out to `strlen32((PCRE2_SPTR32)nametable + imm2_size)`, which luckily applies the cast only to the first half of the expression.

The STRLEN macro is broken, but it just happens to luck out on this case.

The whole system macros here is really, really hard to get right. I'm glad I'm getting rid of it in my other PR.